### PR TITLE
Save cli tests logs

### PR DESCRIPTION
**Pull request by hackaugusto**
Saving the logs of the ethereum node and the raiden nodes to figure out why some tests are flaky. 

Example: https://circleci.com/gh/raiden-network/raiden/52357#tests/containers/16

### CODE REVIEW

File: raiden/tests/integration/cli/conftest.py [hunk 1]
```diff
@@ -77,7 +98,7 @@ def cli_args(raiden_testchain, removed_args, changed_args, environment_type):
         initial_args["secret_registry_contract_address"],
         "--endpoint-registry-contract-address",
         initial_args["endpoint_registry_contract_address"],
-        "--disable-debug-logfile",
+        f"--debug-logfile-name={base_logfile}",
```
Comment (palango) [inline raiden/tests/integration/cli/conftest.py]: I don't find that option, are you sure it exists?

Comment (palango) [reply raiden/tests/integration/cli/conftest.py]: Ah, nevermind. Should have read one.

File: raiden/tests/utils/smoketest.py [hunk 2]
```diff
@@ -152,14 +148,13 @@ def get_private_key(keystore):
 
 @contextmanager
 def setup_testchain(
-    eth_client: EthClient, free_port_generator: Iterable[Port]
+    eth_client: EthClient, free_port_generator: Iterable[Port], base_datadir, base_logdir: str
```
Comment (palango) [inline raiden/tests/utils/smoketest.py]: type annotation for `base_datadir` missing

File: raiden/ui/cli.py [hunk 3]
```diff
@@ -310,10 +310,15 @@ def options(func):
                 "--log-file",
                 help="file path for logging to file",
                 default=None,
-                type=str,
+                type=click.Path(dir_okay=False, writable=True, resolve_path=True),
                 show_default=True,
             ),
             option("--log-json", help="Output log lines in JSON format", is_flag=True),
+            option(
+                "--debug-logfile-name",
```
Comment (palango) [inline raiden/ui/cli.py]: Should we hide this by default? I guess users shouldn't touch this.

Comment (hackaugusto) [reply raiden/ui/cli.py]: why not? if they can use `--disable-debug-logfile` I don't see why they can't move it around.

**Pull request merged**